tasks/xgettext.js

```diff
@@ -302,13 +319,47 @@ module.exports = function(grunt) {
         extractStrings('"', func + "_");
       });
 
-      return messages;
+      return [messages, messageToFilesMap];
     }
   };
 
+  function combineMessageToFilesMaps(mapToModify, newMap) {
+    _.each(newMap, function (files, message) {
+      if (mapToModify[message]) {
+        Array.prototype.push.apply(mapToModify[message], files);
+      } else {
+        mapToModify[message] = files;
+      }
+    });
+  }
+
+  function calculateDiffBetweenOldAndNewMessageIds(oldMessageIdsArray, messageToFilesMap) {
+    var diff = '';
+    _.each(messageToFilesMap, function (files, messageId) {
+      if (!_.contains(oldMessageIdsArray, escapeString(messageId))) {
+        diff += escapeString(messageId) + ' -- ' + files + '\n';
+      }
+    });
+    return diff;
+  }
+
+  function extractMessageIds(string) {
+    var messageIds = [];
+    var match = true;
+    var regex = new RegExp('msgid "((.|[\\s])+?([^\\\\]))"', 'gm');
```

Comment:
new RegExp shouldn't be used unless the regular expression includes a runtime component - it is very easy to get the escaping wrong when using the constructor - instead use a regular expression literal.